perfscale_regression_ci/scripts/encryption/toggle_encryption.sh

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
+###########################################################################################################
+## Auth=skordas@redhat.com
+## Desription: Script to toggle between encryption and decryption of cluster.
+## Polarion test case: OCP-26194 - Compare project loading time with etcd-encryption enabled and disabled
+## https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-26194
+## Cluster config: default
+## Usage: ./toggle_encryption.sh <enable/disable> <timeout_in_minutes>
+## To enable: ./toggle_encryption.sh enable 50
+## To disable: ./toggle_encryption.sh disable 50
+###########################################################################################################
+
+function log {
+    echo -e "$*"
+}
+
+# Setting variables and names of new created secrets.
+mode=""
+timeout=$(($2 * 60))
+## Temporary files to store secrets
+OC_SECRET="/tmp/openshift-apiserver-secret.yaml"
+KB_SECRET="/tmp/openshift-kube-apiserver-secret.yaml"
+OAUTH_SECRET="/tmp/openshift-oauth-apiserver-secret.yaml"
+OC_MIGRATED_RESOURCES="'{\"resources\":[{\"Group\":\"route.openshift.io\",\"Resource\":\"routes\"}]}'"
+## Resources A and B are the the same json objects, but look different as a strings (in check_secret I'm comparing string from yaml)
+KB_MIGRATED_RESOURCES_A="'{\"resources\":[{\"Group\":\"\",\"Resource\":\"configmaps\"},{\"Group\":\"\",\"Resource\":\"secrets\"}]}'"
+KB_MIGRATED_RESOURCES_B="'{\"resources\":[{\"Group\":\"\",\"Resource\":\"secrets\"},{\"Group\":\"\",\"Resource\":\"configmaps\"}]}'"
+OAUTH_MIGRATED_RESOURCES_A="'{\"resources\":[{\"Group\":\"oauth.openshift.io\",\"Resource\":\"oauthaccesstokens\"},{\"Group\":\"oauth.openshift.io\",\"Resource\":\"oauthauthorizetokens\"}]}'"
+OAUTH_MIGRATED_RESOURCES_B="'{\"resources\":[{\"Group\":\"oauth.openshift.io\",\"Resource\":\"oauthauthorizetokens\"},{\"Group\":\"oauth.openshift.io\",\"Resource\":\"oauthaccesstokens\"}]}'"
+## Getting number of existing secrets.
+oc_secrets_num=$(oc get secrets -n openshift-config-managed | grep -c encryption-key-openshift-apiserver)
+kb_secrets_num=$(oc get secrets -n openshift-config-managed | grep -c encryption-key-openshift-kube-apiserver)
+oauth_secrets_num=$(oc get secrets -n openshift-config-managed | grep -c encryption-key-openshift-oauth-apiserver)
+## Setting names for secrets created during encryption/decryption.
+((oc_secrets_num = oc_secrets_num + 1))
+((kb_secrets_num = kb_secrets_num + 1))
+((oauth_secrets_num = oauth_secrets_num + 1))
+oc_secrets_name=$(echo -e "encryption-key-openshift-apiserver-${oc_secrets_num}")
+kb_secrets_name=$(echo -e "encryption-key-openshift-kube-apiserver-${kb_secrets_num}")
+oauth_secrets_name=$(echo -e "encryption-key-openshift-oauth-apiserver-${oauth_secrets_num}")
+log "Secrets to check:"
+log "${oc_secrets_name}"
+log "${kb_secrets_name}"
+log "${oauth_secrets_name}"
+
+function enable_encryption {
+    mode="aescbc"
```

Comment:
is this variable being used?

Reply:
Yes  - as a global variable L:18 later used for comparison L:80

Reply:
Are you able to parse the value set on line 48 instead of setting a separate variable?

Reply:
I'm using two different functions - enable_encryption and disable_encryption - difference between these two is only in different mode value.
In other way enable/disable is setting global `mode` later this global value is compared by one function `check_secret`